fud/bitstream/gen_xo.tcl

```diff
@@ -22,7 +22,9 @@ set_property sdx_kernel_type rtl [ipx::current_core]
 
 # Declare bus interfaces.
 ipx::associate_bus_interfaces -busif s_axi_control -clock ap_clk [ipx::current_core]
-ipx::associate_bus_interfaces -busif m0_axi -clock ap_clk [ipx::current_core]
+foreach busname $argv {
```

Comment:
Does this skip the first argument (i.e., `xoname`)?

Reply:
Sorry, just saw this comment. Following the documentation for Tcl scripts, because I have already fetched the name using `argv0`, I believe `$argv` captures only the remaining command-line arguments.

Reply:
Hmm, I'm not sure about that… I think `$::argv0` might be the zeroth argument, which is the executable, but the first element of `$argv` would be the `$xoname` variable defined above. Anyway, it seems like the thing to do is to test it and see what happens, inserting print statements as necessary—have you tested this out yet?

Reply:
I guess the `lindex` does not show a consuming behavior since it was failing the test. I think I would have to have the arguments be passed in as a list (i.e., "{1, 2, 3, 4, 5}")?

Reply:
OK! To diagnose, might I suggest printing stuff out? That could give you more insight than the binary “test passes or fails” output.

It should be possible to just iterate over the relevant arguments. The equivalent of this Python:

```py
xoname = argv.pop(0)
for arg in argv: ...
```

Or:

```py
for arg in argv[1:]: ...
```

Or:

```py
for (i, arg) in enumerate(argv):
   if i > 0: ...
```

Or whatever, something like that.

Doing this will just require googling how to do something like this in Tcl. Instead of working in this file directly, you can try writing a standalone Tcl script that just iterates over arguments and prints stuff. Then, once you have a loop that works, you can change the “real” script to do the same thing.